spec/registry.md

```diff
@@ -255,3 +255,548 @@ which only expects the `accord` option:
+> one  {{{$numDays} den}}
+> few  {{{$numDays} dny}}
+> many {{{$numDays} dne}}
+> *    {{{$numDays} dní}}
+> ```
+> Using the rules found above, the results of various `operand` values might look like:
+> | Operand value | Keyword | Formatted Message |
+> |---|---|---|
+> | 1 | `one` | 1 den |
+> | 2 | `few` | 2 dny |
+> | 5 | `other` | 5 dní |
+> | 22 | `few` | 22 dny |
+> | 27 | `other` | 27 dní |
+> | 2.4 | `many` | 2,4 dne |
+
+#### Determining Exact Literal Match
+
+> [!IMPORTANT]
+> The exact behavior of exact literal match is only defined for non-zero-filled
+> integer values.
+> Annotations that use fraction digits or significant digits might work in specific
+> implementation-defined ways.
+> Users should avoid depending on these types of keys in message selection.
+
+
+Number literals in the MessageFormat 2 syntax use the 
+[format defined for a JSON number](https://www.rfc-editor.org/rfc/rfc8259#section-6).
+The resolved value of an `operand` exactly matches a numeric literal `key`
+if, when the `operand` is serialized using the format for a JSON number,
+the two strings are equal.
+
+> [!NOTE]
+> Implementations are not expected to implement this exactly as written,
+> as there are clearly optimizations that can be applied.
+
+> [!NOTE]
+> Only integer matching is required in the Technical Preview.
+> Feedback describing use cases for fractional and significant digits-based
+> selection would be helpful.
+> Otherwise, users should avoid using matching with fractional numbers or significant digits.
+
+## Date and Time Value Formatting
+
+This subsection describes the functions and options for date/time formatting.
+Selection based on date and time values is not required in this release.
+
+### Functions
+
+Functions for formatting [date/time values](#operands) in the default registry are:
+
+- `:datetime`
+- `:date`
+- `:time`
+
+If no options are specified, each of the functions defaults to the following:
+- `{$d :datetime}` is the same as `{$d :datetime dateStyle=short timeStyle=short}`
+- `{$d :date}` is the same as `{$d :date style=short}`
+- `{$t :time}` is the same as `{$t :time style=short}`
+
+> [!NOTE]
+> The default formatting behavior of `:datetime` is inconsistent with `Intl.DateTimeFormat`
+> in JavaScript and with `{d,date}` in ICU MessageFormat 1.0.
+> This is because, unlike those implementations, `:datetime` is distinct from `:date` and `:time`.
+
+### Operands
+
+The _operand_ of a date/time function is either 
+an implementation-defined date/time type (passed in as an argument)
+or a _date/time literal value_, as defined below.
+All other _operand_ values produce a _Selection Error_.
+
+A **_<dfn>date/time literal value</dfn>_** is a non-empty string consisting of 
+one of the following:
+- an XMLSchema 1.1 [dateTime](https://www.w3.org/TR/xmlschema11-2/#dateTime)
+- an XMLSchema 1.1 [time](https://www.w3.org/TR/xmlschema11-2/#time)
+- an XMLSchema 1.1 [date](https://www.w3.org/TR/xmlschema11-2/#date)
```

Comment:
We should clarify which functions take which of these values. All can't be valid for all, as e.g. an XML time is insufficient input for `:date`.

Also, are any timezones in literal inputs used in the formatted outputs? For example, when formatting
```
{12:34+06:00 :time}
```
is the resulting formatted time in the system timezone, or in +06?

How about UTC, what timezone does this render in?
```
{12:34Z :time}
```

As we've left out `timeZone` from the available options, also leaving out timezone parsing would be one way to resolve this.

Reply:
Oh boy, where do I start? 😈 

The values can parse into various kinds of date/time objects. Many (most?) of these can be formatted by any of the functions given here, although the results may be not be very useful. Specifically, a `time` formatted as a date or a `date` formatted as a time is probably not very interesting.

XMLSchema date/time values _do not have timezones_. They only support _zone offsets_. The only zone offset that can be 100% certainly made into a time zone is `Z` because that one means "UTC".

The time zone is very important in rendering date and time values. I argued at length in the previous PR (and tried to make the point in the call) that we should not excommunicate `timezone` as an option to "future". Without that option, there are only the following options for managing time zone in formatting:

- the implementation uses the system default
- the implementation provides an API for setting the time zone of an entire message
  (this is not a behavior we specify because it is beyond our remit, although probably
  we should point it out)
- the implementation assumes UTC

In addition, date/time _literals_ with no offset should be handled as floating time values (even if the implementation uses some specific time zone, the value should be formatted so that the field values don't change--often this is done using UTC as the time zone, ignoring local settings)

Note well that "time zone parsing" doesn't exist here. The _zone offset_ (usually include `Z`) parses into the underlying incremental ("instant") value. That is, all of the following are the *same* time value:

- `2007-01-01T01:00:00.000-01:00`
- `2007-01-01T01:00:00.000Z` 
- `2007-01-01T01:00:00.000+01:00` 
- `2006-12-31T23:00:00.000-01:00`

Literals missing one part should be permitted to treat that part as `0`. Some examples:

| Literal | Pattern | Output | Comment |
|---|---|---|---|
| `2024-02-17` | `{$d :datetime dateStyle=medium}` | Feb 17, 2024 00:00 a.m. | No time; treat as floating |
| `12:34:56.000` | {$d :datetime dateStyle=medium}` | Jan 1, 1970 12:34 p.m. | No date; treat as floating |
| `2024-02-17T12:34:56.000Z` | {$d :dateTime dateStyle=medium} | Feb 17, 2024 4:34:56 a.m. Pacific | Surprised? |
| `2024-02-17T12:34:56.000-08:00` | {$d :dateTime dateStyle=medium} | Feb 17, 2024 12:34 p.m. Pacific | |

If we didn't do this, users used to working with partial values would have messages fail instead of doing what their code usually does. The group will need to decide if we bring `timeZone` back in Tech Preview. We should also discuss whether to specify any of this behavior.

Some of what I express above might be controversial, in that we could make different decisions about forcing floating time behavior or permitting partial values.

Reply:
Note to @eemeli... see disconnected conversation below ⬇️